### PR TITLE
Created new MultipleChoiceSurveyQuestionDialog

### PR DESCRIPTION
**Pull request by cforkish**
**before:**
Screen Shot 2020-07-20 at 2 04 44 PM

**after:**
Screen Shot 2020-07-20 at 2 05 17 PM

previously, clicking on a question in the survey overview table would open the `MultipleChoiceDetailsDialog`, which wasn't designed to fetch or display survey data. the easiest solution was to create a new `MultipleChoiceSurveyQuestionDialog` (largely copied from the previous dialog) which is designed specifically to display details for a multiple choice survey question.

 ### Future work
This code needs cleanup: [jira](https://codedotorg.atlassian.net/browse/LP-1524)

## Links

- [jira](https://codedotorg.atlassian.net/browse/LP-1505)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyOverviewTable.jsx
Comment (jmkulwik) [inline apps/src/templates/sectionAssessments/MultipleChoiceSurveyOverviewTable.jsx:237]: NIT: an (optional) shorthand for this is `const {selectedQuestionIndex} = this.state`

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (jmkulwik) [inline apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:58]: NIT (optional cleanup): access `this.props.isDialogOpen` and `this.props.closeDialog` (here and on line 90) the same way you access questionData at the top of the render function. i.e. `const {questionData, isDialogOpen, closeDialog} = this.props;`

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (jmkulwik) [inline apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: Is it possible to avoid `__useDeprecatedTag`? 
This param causes Button to use a `<div>` instead of a `<button>` element.

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (jmkulwik) [inline apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:65]: Worth it to add a test for this?

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyOverviewTable.jsx
Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyOverviewTable.jsx:237]: lol. if i use that shorthand, the next line would wrap. so i wanted to give the value a shorter name while assigning. but if we like to be consistent about using that shorthand (which doesn't seem to be the case as far as i can tell) i'm happy to update this.

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:58]: i admit that i copy/pasted most of this code from `MultipleChoiceDetailsDialog`, and didn't spend any time looking for places to cleanup the copied code. my gut instinct was to factor out the common parts from both components so as not to duplicate code, but i realized that was a slippery slope leading to an audit of how we compose anything using `BaseDialog` so i decided to punt and opt for the quick and dirty solution until i'm more familiar with our UI codebase. since you mark this as "optional cleanup" i'm going to leave it for now, since if i clean it up here i would feel obligated to clean it up in `MultipleChoiceDetailsDialog` as well, and... well, another slippery slope. tl;dr let's put cleaning up our dialog composition code on the TODO list!

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: tbh i didn't even bother learning what that tag did, thinking consistency with other similar dialogs was good enough for me. see above about slippery slopes. if @clareconstantine thinks it's worth the time, i'm happy to slide down those slopes and use these two dialogs to develop some higher levels of abstraction on top of `BaseDialog` so we don't have duplicated code in these two components. if not though, i think keeping it consistent with the other dialog would be better than cleaning up just one.

thanks for explaining what this tag does though! i'll need to do some reading to understand the implications, but good to know the gist of it!

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:65]: probably! what specifically are you referring to? or just adding some testing for this component in general?

Comment (jmkulwik) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:65]: General testing would be great! Specifically, however, I was referring to ensuring the correct number of answers render (or don't render).

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (jmkulwik) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: We generally try to use idiomatic HTML where possible on new elements for accessibility purposes and WCAG compliance. It's an area we have been historically bad at, but via efforts to deprecate a few non-compliant elements this year, we are starting to make progress. For this particular deprecated tag, we'd generally want a pretty significant reason to keep it here (clickable divs are a focus for accessibility work this year). If you do decide to leave it, please create an item on the backlog so we can come back and clean it up. 

In general for features/elements/modules we've deprecated, we take a "clean up as you go" approach. It spreads the work out across the entire engineering team so we can eventually fully remove them without a large disruption.

Comment (mvkski) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: I second @jmkulwik here. It'd be good to clean this up.

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (mvkski) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:65]: Seconding @jmkulwik

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (mvkski) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:58]: > my gut instinct was to factor out the common parts from both components so as not to duplicate code

I like your base instincts. I strongly, strongly support taking the extra time to make our changes as high quality as possible

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (mvkski) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: Also in general I recommend taking time to understand code written for PRs - up to a limit of course, since its generally not possible to understand how every detail of what you write works - but generally speaking, understanding what the parameters used in all of the functions you call do falls within the scope of what I expect someone writing a diff to be responsible for in the large majority of cases. Is there an aggressive deadline on this or other tasks that is causing this to be difficult?

Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:88]: right on, i'm more than happy to spend time cleaning this up. and good point about understanding any code that i'm reusing @mvkski – in retrospect this was a very lazy approach. stay tuned!

File: apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx
Comment (cforkish) [reply apps/src/templates/sectionAssessments/MultipleChoiceSurveyQuestionDialog.jsx:65]: 👍

**Comment by cforkish** [comment]
okay, this turned out to be a steep and slippery slope indeed. we have _a lot_ of duplicate code around showing assessment/survey response data so the best approach would be to clean it all up at once. luckily i got approval from Clare and Amanda to tackle this cleanup, but i have one higher priority issue i need to handle first. i'm closing this PR because it will be irrelevant in the context of the broader cleanup.

**Comment by cforkish** [comment]
the task of cleaning up all the duplicated code related to this PR has been pushed in favor of tasks that will have a more immediate impact for teachers starting the school year, so @clareconstantine and i decided we should go ahead and merge this since it solves a live problem. i will revisit the comments here when i tackle the larger cleanup task.

**Review by mvkski — APPROVED** [review]
Happy to approve despite the outstanding issues as this solves a live problem. I reccomend linking a follow up jira item to this PR to ensure we get back to it.

**Review by jmkulwik — APPROVED** [review]
I agree with Max! Approving so we can get the fix out, but please add a jira ticket for the cleanup!

One low-hanging fruit that would be great if it could sneak in as part of this PR is the removal of `__useDeprecatedTag`. If removing it causes problems, go ahead and leave it, but it should be able to be deleted without any noticeable difference.

**Comment by cforkish** [comment]
good call @mvkski: https://codedotorg.atlassian.net/browse/LP-1524

@jmkulwik i've noticed we have that tag in _a lot_ of the buttons i've seen in our code, so i would like to take some time to understand why that is before just removing it. i will look into that when i tackle [LP-1524], but we might even want to separate it out into its own task and handle it everywhere all at once instead of just in this assessment response code.

[LP-1524]: https://codedotorg.atlassian.net/browse/LP-1524

**Pull request merged**
